modules/backend/assets/ui/js/ui/WidgetHandler.js

```diff
@@ -0,0 +1,70 @@
+/**
+ * Backend widget handler.
+ *
+ * Handles the creation and disposal of widgets in the Backend. Widgets should include this as
+ * a dependency in order to be loaded and initialised after the handler, in order to correctly
+ * register.
+ *
+ * @copyright 2021 Winter.
+ * @author Ben Thomson <git@alfreido.com>
+ */
+export default class WidgetHandler extends Snowboard.Singleton {
+    construct() {
+        this.registeredWidgets = [];
+    }
+
+    listens() {
+        return {
+            ready: 'onReady',
+            ajaxUpdate: 'onAjaxUpdate',
+        };
+    }
+
+    register(control, widget, callback) {
+        this.registeredWidgets.push({
+            control,
+            widget,
+            callback,
+        });
+    }
+
+    unregister(control) {
+        this.registeredWidgets = this.registeredWidgets.filter((widget) => widget.control !== control);
+    }
+
+    onReady() {
+        this.initialiseWidgets(document.body);
+    }
+
+    onAjaxUpdate(element) {
+        this.initialiseWidgets(element);
+    }
+
+    initialiseWidgets(element) {
```

Comment:
We should stick to the US spelling of initialize: https://github.com/search?q=org%3Awintercms+initialize&type=code vs https://github.com/search?q=org%3Awintercms+initialise&type=code